python_modules/dagster/dagster/_core/definitions/declarative_automation/automation_condition.py

```diff
@@ -868,8 +868,17 @@ def _get_stable_unique_id(self, target_key: Optional[EntityKey]) -> str:
         This should only be used for conditions that don't change their output based on what conditions are
         evaluated before them (i.e. they explicitly set their candidate subset to the entire subset).
         """
+        from dagster._core.definitions.declarative_automation.operators import (
+            NewlyTrueCondition,
+            SinceCondition,
+        )
+
         child_ids = [
-            child.get_node_unique_id(
+            child._get_stable_unique_id(target_key)  # noqa
+            if isinstance(
+                child, (SinceCondition, NewlyTrueCondition)
+            )  # temporary forwards compat hack
```

Comment:
just showing my work here on how i gained understanding here

for this to be forwards-compat, _get_stable_unique_id needs to be, well, stable, before and after the code change

For conditions whose IDs are changing as part of https://github.com/dagster-io/dagster/pull/32856, get_node_unique_id changes, so _get_stable_unique_id changes, so we lose the forward-compat guarantees we were looking for

now, _get_stable_unique_id will not change as a result of that change when it rolls out again

And once it is rolled out, this if will be meaningfless, since get_node_unique_id will now always equal _get_stable_unique_id for thos conditions anyway, so we can remove it since it will be a no-op

Phew!